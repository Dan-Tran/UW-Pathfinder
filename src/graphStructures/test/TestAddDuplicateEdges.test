# This will test adding nodes to the graph.

# Create the graph.
CreateGraph graphtest

# Make sure the graph has no nodes and no edges.
ListNodes graphtest

# Add nodes
AddNode graphtest n1
AddNode graphtest n2
AddNode graphtest n3
AddNode graphtest n4

# Make sure the nodes were added
ListNodes graphtest

# Add an edge between n1 and n2
AddEdge graphtest n1 n2 e1

# Make sure the edge was added
ListChildren graphtest n1

# Make sure n2 is not a parent.
ListChildren graphtest n2

# Add duplicate edges
AddEdge graphtest n1 n2 e1
AddEdge graphtest n1 n2 e2
AddEdge graphtest n2 n1 e2
AddEdge graphtest n3 n2 e1
AddEdge graphtest n3 n1 e2
AddEdge graphtest n3 n4 e2

# Make sure graph is correct.
ListNodes graphtest

ListChildren graphtest n1
ListChildren graphtest n2
ListChildren graphtest n3