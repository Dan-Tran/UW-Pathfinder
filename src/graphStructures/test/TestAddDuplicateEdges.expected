# This will test adding nodes to the graph.

# Create the graph.
created graph graphtest

# Make sure the graph has no nodes and no edges.
graphtest contains: 

# Add nodes
added node n1 to graphtest
added node n2 to graphtest
added node n3 to graphtest
added node n4 to graphtest

# Make sure the nodes were added
graphtest contains: n1 n2 n3 n4

# Add an edge between n1 and n2
added edge e1 from n1 to n2 in graphtest

# Make sure the edge was added
the children of n1 in graphtest are: n2(e1)

# Make sure n2 is not a parent.
the children of n2 in graphtest are:

# Add duplicate edges
added edge e1 from n1 to n2 in graphtest
added edge e2 from n1 to n2 in graphtest
added edge e2 from n2 to n1 in graphtest
added edge e1 from n3 to n2 in graphtest
added edge e2 from n3 to n1 in graphtest
added edge e2 from n3 to n4 in graphtest

# Make sure graph is correct.
graphtest contains: n1 n2 n3 n4

the children of n1 in graphtest are: n2(e1) n2(e2)
the children of n2 in graphtest are: n1(e2)
the children of n3 in graphtest are: n1(e2) n2(e1) n4(e2)